dist/tools/openocd/openocd.sh

```diff
@@ -178,6 +178,13 @@ do_flash() {
 do_debug() {
     test_config
     test_elffile
+
+    # Configure halt command: prepend a reset if required by the board config
+    local _pre_reset_halt=''
+    if [ -n "${PRE_DEBUG_RESET_HALT}" ]; then
```

Comment:
For me, the `PRE` here is wrong, because it changes the command from `halt` to `reset halt` it's not a previous command. Also, introducing a new really specific variable only for this specific way of handling the case seems bad to me.

Reply:
I would prefer the following

    # A meaningful comment on what this is :p
    : ${OPENOCD_EXTRA_INIT:=-c 'halt'}

And in the given board set

    export OPENOCD_DEBUG_HALT_CMD = -c 'reset halt'